setup.sh

```diff
@@ -2,8 +2,26 @@
 
 CURRENT_PATH=$(pwd)
 ALIAS_COMMAND="alias git-pr-local='sh $CURRENT_PATH/git-pr-local.sh'"
+ALIAS_COMMAND_REGEX="alias git-pr-local.*"
 
-echo "\n$ALIAS_COMMAND" >> ~/.bashrc
-echo "\n$ALIAS_COMMAND" >> ~/.bash_profile
-echo "\n$ALIAS_COMMAND" >> ~/.profile
-echo "\n$ALIAS_COMMAND" >> ~/.zshrc
+declare -a SUPPORTED_TERMINAL_FILES=(".bashrc" ".bash_profile" ".profile" ".zshrc")
+
+# Run the command in current context
+eval $ALIAS_COMMAND
+
+for file in "${SUPPORTED_TERMINAL_FILES[@]}"
+do
+	# Only try adding alias to profiles when the file is readable && writable
+	if [ -w ~/"$file" ] && [ -r ~/"$file" ]
+	then
+		# Remove older added aliases before adding the new ones
+		sed -i "s/$ALIAS_COMMAND_REGEX//g" ~/"$file" 2>/dev/null
+		# For MacOS, `sed -i` expects '' argument, otherwise fails with weird errors
+		sed -i '' "s/$ALIAS_COMMAND_REGEX//g" ~/"$file" 2> /dev/null
+
+		# Add the alias
+		/bin/echo -n $ALIAS_COMMAND >> ~/$file
```

Comment:
@hemal7735 how does it look/work now? If you're near a linux machine, could you help test it out?

Reply:
Do we need `/bin/echo` ? earlier it was just `echo`, right ?

Reply:
and `\n` is needed before the `ALIAS_COMMAND`, as we don't know if there is new-line already in the files

Reply:
Yes. We do need `/bin/echo` since there is a `builtin` echo and a command echo. The one used by default in mac might not understand `-n` (at least for me it didn't).

For me, it auto-added the `\n` even before the alias command, so I removed it. Can you test it out with your terminal and/or a linux terminal (if you're near one)?

Reply:
I don't have linux terminal, but I'm using oh-my-zsh and `echo -n` worked in that terminal. 
`/bin/echo -n` worked as well.
I think we can merge now.

Reply:
Sounds good. Could you please merge it?